File: 백준/Bronze/25304. 영수증/영수증.py
import sys

readline = sys.stdin.readline

x = int(readline())
n = int(readline())
a = b = 0

for _ in range(0, n) :
    a, b = map(int, readline().split())

    x -= a * b

print("Yes" if x == 0 else "No", end = "")

File: 백준/Bronze/25304. 영수증/영수증.js
const fs = require('fs');

const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

let x = input[0];
const n = input[1];

for (let i = 0; i < n; i++) {
    const [a, b] = input[2 + i].split(' ').map(Number);

    x -= a * b;
}

console.log(x == 0 ? "Yes" : "No");